python312Packages.enlighten: 1.13.0 -> 1.14.1

doronbehar opened the pull request:
## Description of changes


## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

doronbehar committed:
python312Packages.enlighten: no with lib; in meta

doronbehar committed:
python312Packages.enlighten: add doronbehar as co-maintainer

doronbehar committed:
python312Packages.enlighten: 1.13.0 -> 1.14.0

Changelog: https://github.com/Rockhopper-Technologies/enlighten/releases/tag/1.14.0

doronbehar committed:
python312Packages.enlighten: reorder inputs & modernize attributes

ethancedwards8 commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 388674`

---
### `aarch64-darwin`
<details>
  <summary>:x: 8 packages failed to build:</summary>
  <ul>
    <li>nsz (python312Packages.nsz)</li>
    <li>nsz.dist (python312Packages.nsz.dist)</li>
    <li>python312Packages.enlighten</li>
    <li>python312Packages.enlighten.dist</li>
    <li>python313Packages.enlighten</li>
    <li>python313Packages.enlighten.dist</li>
    <li>python313Packages.nsz</li>
    <li>python313Packages.nsz.dist</li>
  </ul>
</details>

ethancedwards8 commented:
```
error: builder for '/nix/store/x0kbm8437lf6fdagy22hrrbd5r8m34rz-python3.12-enlighten-1.14.0.drv' failed with exit code 1;
       last 25 log lines:
       >         self.tty.clear()
       >
       >         # Counter 1 in auto-refresh list
       >         self.assertIn(counter1, manager.autorefresh)
       >
       >         # If auto-refreshed counter hasn't been refreshed recently, refresh
       >         counter1.last_update = 0
       >         counter2.refresh()
       >         self.tty.stdout.write(u'X\n')
       >         output = self.tty.stdread.readline()
       >         self.assertRegex(output, 'counter2.+counter1')
       >
       >         # If auto-refreshed counter has been refreshed recently, skip
       >         counter1.last_update = counter1.start + 5.0
       >         counter2.refresh()
       >         self.tty.stdout.write(u'X\n')
       >         output = self.tty.stdread.readline()
       >         self.assertRegex(output, 'counter2')
       > >       self.assertNotRegex(output, 'counter1')
       > E       AssertionError: Regex matched: 'counter1' matches 'counter1' in '\x1b[6n\x1b[25;1H\r\x1b[Kcounter2\x1b[24;1H\r\x1b[Kcounter1\x1b[23;1HX\n'
       >
       > tests/test_manager.py:384: AssertionError
       > =========================== short test summary info ============================
       > FAILED tests/test_manager.py::TestManager::test_autorefresh - AssertionError: Regex matched: 'counter1' matches 'counter1' in '\x1b[6n\x1...
       > ============= 1 failed, 136 passed, 1 skipped in 92.83s (0:01:32) ==============
       For full logs, run 'nix log /nix/store/x0kbm8437lf6fdagy22hrrbd5r8m34rz-python3.12-enlighten-1.14.0.drv'.
error: 1 dependencies of derivation '/nix/store/57zzka2s27j8wqdq22kkffb566ciy235-python3.12-nsz-4.6.1.drv' failed to build
error: builder for '/nix/store/qx0fn0wzms2swka07hc6mzw18zv2qn9z-python3.13-enlighten-1.14.0.drv' failed with exit code 1;
       last 25 log lines:
       >         self.tty.clear()
       >
       >         # Counter 1 in auto-refresh list
       >         self.assertIn(counter1, manager.autorefresh)
       >
       >         # If auto-refreshed counter hasn't been refreshed recently, refresh
       >         counter1.last_update = 0
       >         counter2.refresh()
       >         self.tty.stdout.write(u'X\n')
       >         output = self.tty.stdread.readline()
       >         self.assertRegex(output, 'counter2.+counter1')
       >
       >         # If auto-refreshed counter has been refreshed recently, skip
       >         counter1.last_update = counter1.start + 5.0
       >         counter2.refresh()
       >         self.tty.stdout.write(u'X\n')
       >         output = self.tty.stdread.readline()
       >         self.assertRegex(output, 'counter2')
       > >       self.assertNotRegex(output, 'counter1')
       > E       AssertionError: Regex matched: 'counter1' matches 'counter1' in '\x1b[6n\x1b[25;1H\r\x1b[Kcounter2\x1b[24;1H\r\x1b[Kcounter1\x1b[23;1HX\n'
       >
       > tests/test_manager.py:384: AssertionError
       > =========================== short test summary info ============================
       > FAILED tests/test_manager.py::TestManager::test_autorefresh - AssertionError: Regex matched: 'counter1' matches 'counter1' in '\x1b[6n\x1...
       > ============= 1 failed, 136 passed, 1 skipped in 92.77s (0:01:32) ==============
       For full logs, run 'nix log /nix/store/qx0fn0wzms2swka07hc6mzw18zv2qn9z-python3.13-enlighten-1.14.0.drv'.
error: 1 dependencies of derivation '/nix/store/pyp44ngswkjgci8563m75p1fqs20vh3r-python3.13-nsz-4.6.1.drv' failed to build
error: 4 dependencies of derivation '/nix/store/zqmvmwpqygdj20s43q8k02pqw87zxg86-review-shell.drv' failed to build
```

doronbehar commented:
Thanks :)

- https://github.com/Rockhopper-Technologies/enlighten/issues/74

doronbehar committed:
python312Packages.enlighten: 1.14.0 -> 1.14.1

Changelog: https://github.com/Rockhopper-Technologies/enlighten/releases/tag/1.14.1

doronbehar commented:
Merged as CI was completely green.
